src/OmniSharp.MSBuild/ProjectLoader.cs

```diff
@@ -110,7 +110,11 @@ private MSB.Evaluation.Project EvaluateProjectFileCore(string filePath)
 
             toolsVersion = GetLegalToolsetVersion(toolsVersion, projectCollection.Toolsets);
 
-            return projectCollection.LoadProject(filePath, toolsVersion);
+            var project = projectCollection.LoadProject(filePath, toolsVersion);
+
+            SetTargetFrameworkIfNeeded(project);
```

Comment:
Every project goes through this code path, where we first evaluate the project with MSBuild. At this point, we need to set the `<TargetFramework/>` property if there are multiple `<TargetFrameworks/>` to ensure that the project evaluates with a TFM. Otherwise, several MSBuild properties (like `TargetFrameworkMoniker`) are missing.